addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1539,28 +1539,66 @@ options.registry.Header = options.Class.extend({
     },
 });
 
-options.registry.topMenuTransparency = options.Class.extend({
+options.registry.TopMenuVisibility = options.Class.extend({
+    /**
+     * @override
+     */
+    async start() {
+        await this._super(...arguments);
+        const shown = await this._isShown();
+        this.trigger_up('snippet_option_visibility_update', {show: shown});
+    },
```

Comment:
For these feature, the logic is already duplicated: an xml template adds some classes to hide and we simulate them in JS. It's not great but ok while we don't have a way to reload the page fast.

However, in your previous code, you duplicated the logic again by removing the classes here in the start... always try not to duplicate logic.

I changed your code a lot for this feature to take into account async part and share code.